exp/runtime/topologymutation/variables.go

```diff
@@ -53,3 +56,46 @@ func GetStringVariable(templateVariables map[string]apiextensionsv1.JSON, variab
 	}
 	return stringValue, true, nil
 }
+
+// GetBoolVariable get the value as a bool.
+func GetBoolVariable(templateVariables map[string]apiextensionsv1.JSON, variableName string) (bool, bool, error) {
```

Comment:
Not sure it should be a blocker here but `bool, book, error` as a return value seems pretty confusing and easy to get wrong.

I think we should either have a custom result type, or else a custom error which will allow us to check for the `variableNotFound` case. I think the custom error is probably the better option and matches what we do elsewhere.

We should adopt the same pattern for the other `Get*Variable` functions.

Reply:
The GetVariable already implements a wrapper around not found error
```
https://github.com/kubernetes-sigs/cluster-api/blob/main/exp/runtime/topologymutation/variables.go#L33-L37
```
I'm ok to change them all to return error directly.